assets/js/sqoot.js

```diff
@@ -1,254 +0,0 @@
-
-
-$("form").on("submit", function(event) {
-  event.preventDefault();
-
-  $('#deal-table-panel').remove();
-
-  //Get items from input form and assign them to global variables
-  userSearch = $(".user-search-term").val().trim();
-  userLocation = $(".user-location").val().trim();
-
-  //Console log the two search terms to make sure they are being retrieved from the UI
-  console.log(userSearch);
-  console.log(userLocation);
-
-  var webResults = false;
-  var results = 5;
-
-  //Set the UI input fields back to blank for the user
-  $(".user-search-term").val('');
-  $(".user-location").val('');
-
-  //Start building the elements inside the deal-table-row, including panel, and table that will contain the deal data
-  var boostrapTablePanel     = $('<div class="panel panel-default" id="deal-table-panel">'),
-      boostrapTablePanelBody = $('<div class="panel-body" id="deal-table-panel-body">');
-  var tableWhole = $('<table class="table table-striped" id="main-deal-table"><tbody></tbody></table>')
-  var tableHeaderRow = $('<tr id="table-header-row">')
-
-  //Build the table header
-  var tableHeader0 = $('<th>').text('Favorite');
-  var tableHeader1 = $('<th>').text('Business Name');
-  var tableHeader2 = $('<th>').text('Deal');
-  var tableHeader3 = $('<th class="centerText">').text('Deal Price');
-  var tableHeader4 = $('<th class="centerText">').text('Original Value');
-  var tableHeader5 = $('<th class="centerText">').text('Discount');
-
-  //Append the table panel to the deal-table-row that is on the index page to hold the table
-  $('#deal-table-row').append(boostrapTablePanel);
-  //Append the table panel body to the panel that was just created and appended
-  $('#deal-table-panel').append(boostrapTablePanelBody);
-  //Append the table to the panel body that was just created and appended
-  $('#deal-table-panel-body').append(tableWhole);
-
-  //Fill out the header row with the relevant headers, then append the header to the table
-  tableHeaderRow.append(tableHeader0);
-  tableHeaderRow.append(tableHeader1);
-  tableHeaderRow.append(tableHeader2);
-  tableHeaderRow.append(tableHeader3);
-  tableHeaderRow.append(tableHeader4);
-  tableHeaderRow.append(tableHeader5);
-  $('#main-deal-table').append(tableHeaderRow);
-
-
-  //Use the userLocation to retrieve the latitude and longitude where the user marker should be placed
-  //This will be retrieved using the google geolocation API
-  geocode(userLocation);
-
-  function geocode(searchlocation) {
-    axios.get('https://maps.googleapis.com/maps/api/geocode/json', {
-      params: {
-        address: searchlocation,
-        key: 'AIzaSyAvZV3rec51FwhSWqrrRkCySQw6h0cKfJk'
-      }
-    })
-    .then(function(response) {
-      userlat = response.data.results[0].geometry.location.lat;
-      userlng = response.data.results[0].geometry.location.lng;
-
-      console.log(userlat);
-      console.log(userlng);
-    })
-    .catch(function(error) {
-      console.log(error);
-    });
```

Comment:
axios is a great tool for making http requests and supports promises. However, jQuery also supports these features so I am wondering why axios was included. It is good practice to not include libraries to your code unless you have a good reason to. Interviewers will often ask about what technologies you used in your projects and why.

Reply:
I used axios to get around a CORS issue, after trying several other ways to solve the issue. It would be great to hear about a JQuery solution to this problem for the future.

Reply:
How did you come across the problem? I tried replicating this error, but I didn't get any errors with jQuery.